app/controllers/api/api_ao_messages_controller.rb

```diff
@@ -2,6 +2,12 @@ class ApiAoMessagesController < ApiAuthenticatedController
   before_filter :require_account_and_application!
   include AoMessageCreateCommon
 
+  rescue_from(KeyError) { head :bad_request }
+
+  def index
+    render :json => AoMessage.find_all_by_application_id_and_token(@application.id, params.fetch(:token))
```

Comment:
Would this include pagination?

We usually don't need it, but we've also seen some tokens in prod shared between +2k AOs.

Reply:
I missed your comment @matiasgarciaisaia: there is no pagination for the time being, I didn't think about that.